### PR TITLE
PIM-7263: extract command's logic into a dedicated service

### PR DESCRIPTION
**Pull request by grena**
**Description (for Contributor and Core Developer)**

This is a rework to extract logic of the command into a dedicated service. This is the same work than the backport on 2.0 : https://github.com/akeneo/pim-community-dev/pull/7891

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Y
| Added legacy Behats               | N
| Added acceptance tests            | N
| Added integration tests           | N
| Changelog updated                 | Y
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Command/Cleaner/WrongBooleanValuesOnVariantProductCleaner.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+     * Return true if the product has been modified, false otherwise
+     *
+     * @param ProductInterface $variantProduct
+     *
+     * @return bool
+     */
+    public function cleanProduct(ProductInterface $variantProduct): bool
+    {
+        $isModified = false;
+
```
Comment (damien-carcel) [inline src/Pim/Bundle/CatalogBundle/Command/Cleaner/WrongBooleanValuesOnVariantProductCleaner.php]: I would add an early return here if `$variantProduct->isVariant()` is false. In 2.0, the typehint of the method prevented this case to happen. Not the case here anymore, so anyone could missuse this service.

File: src/Pim/Component/Catalog/spec/Updater/WrongBooleanValuesOnVariantProductUpdaterSpec.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
+<?php
```
Comment (christophe-chausseray) [inline src/Pim/Component/Catalog/spec/Updater/WrongBooleanValuesOnVariantProductUpdaterSpec.php]: This one could be removed as now we have the WrongBooleanValuesOnVariantProductCleanerSpec.php

File: src/Pim/Bundle/CatalogBundle/Command/Cleaner/WrongBooleanValuesOnVariantProductCleaner.php (hunk 1 above)
Comment (christophe-chausseray) [reply src/Pim/Bundle/CatalogBundle/Command/Cleaner/WrongBooleanValuesOnVariantProductCleaner.php]: The check is already done there : https://github.com/akeneo/pim-community-dev/pull/7928/commits/84faa7c9c39719124977dbbba638a3d05beb5893#diff-7e00d45aa78c74e5d86d6db40f4d5316R62

Comment (grena) [reply src/Pim/Bundle/CatalogBundle/Command/Cleaner/WrongBooleanValuesOnVariantProductCleaner.php]: Nice catch, I add it.

**Pull request merged**